### PR TITLE
Clean up conda after each test

### LINKED ISSUE #248: GitHub Actions failure `NoSpaceLeftError: No space left on devices.` with unicycler conda environment creation
**Snakemake version**
snakemake version: ` 5.26.1`
wrapper version: not yet released, current version on master branch

**Describe the bug**
The tests on the pull request for unicycler's addition ran fine (and quick, a few minutes):
https://github.com/snakemake/snakemake-wrappers/runs/1342977411#step:7:265

Since it has been merged into master, tests on master are consistently failing with the following error (after running for hours):
```
Creating conda environment /tmp/tmprc8d9w2k/master/bio/unicycler/environment.yaml...
Downloading and installing remote packages.
CreateCondaEnvironmentException:
Could not create conda environment from /tmp/tmprc8d9w2k/master/bio/unicycler/environment.yaml:
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
Preparing transaction: ...working... done
Verifying transaction: ...working... failed

NoSpaceLeftError: No space left on devices.
```
See here:
https://github.com/snakemake/snakemake-wrappers/runs/1343943714#step:7:321

**Logs**
If applicable, any terminal output to help explain your problem.

**Minimal example**
See GitHub Actions results of recent merge commits.

### PR DESCRIPTION
**Pull request by dlaehnemann**
This pull request is meant to fix #248, but we'll have to test it first, to see if it works.

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -88,8 +88,8 @@ def run(wrapper, cmd, check_log=None):
         finally:
             # cleanup environments to save disk space
             subprocess.check_call(
-                "for env in `conda env list | grep -P '\.snakemake/conda' | "
-                "cut -f1 | tr -d ' '`; do conda env remove --prefix $env; done",
+                "for env in `conda env list | grep '{}' | grep -P '\.snakemake/conda' | "
+                "cut -f1 | tr -d ' '`; do conda env remove --prefix $env; done".format(dst),
```
Comment (mbargull) [inline test.py:92]: FYI, you can use `conda cmd --json` to get more structured data, e.g., something like
```
conda info --env --json | python -c 'import json, sys; print(*json.load(sys.stdin)["envs"], sep="\n")' | grep-here-or-in-python-call | xargs -L1 conda remove --all --yes --prefix
```
would also work (beware: written without testing ;).

Comment (dlaehnemann) [reply test.py:92]: Any advantages or disadvantages over the grep-only variant? Fail-safer or something?

Comment (mbargull) [reply test.py:92]: The JSON output is considered API, the bare text output from `conda`/`conda-env` is not guaranteed to stay the same and can change with any update of `conda`.

**Comment by johanneskoester** [comment]
Closing in favor of the fixes in master. Thanks a lot for all this exploratory work, it helped a lot!

**Pull request closed without merging**